### ISSUE TITLE
New PSF format fails QL

### ISSUE THREAD

**Issue by Srheft** [desihub/desispec#681]
Using the MockObs format of the desi calibration data (located in $DESI_CCD_CALIBRATION_DATA)
QL fails to process arc exposures. See the running example from 20191001 below:

**time desi_quicklook -i ./config_files/qlconfig_arc.yaml -n 20191001 -c b3 -e 3573 --plots**
QuickLook CRITICAL : Failed to run PA ResolutionFit error was no ysig coefficents were read in the PSF
Traceback (most recent call last):
  ...
  File "/desi/software/desispec/py/desispec/quicklook/qlpsf.py", line 130, in ysigma
    return self.traceset.ysig_vs_wave(ispec,wave)
  File "/desi/software/desispec/py/desispec/xytraceset.py", line 61, in ysig_vs_wave
    raise RuntimeError("no ysig coefficents were read in the PSF")
RuntimeError: no ysig coefficents were read in the PSF
Failed to run PA ResolutionFit

**Comment by julienguy** [comment]
Can you please provide more details for me to reproduce the bug at NERSC.
 - path to the data
 - env. variables
 - code version (master?)

**Comment by sbailey** [comment]
NERSC is down today and tomorrow, but I think the relevant files are also at KPNO.  @Srheft can you provide cut-and-paste instructions for how to reproduce the problem starting with a fresh login of quicklook@desi-1.kpno.noao.edu?  Thanks.

**Comment by rstaten** [comment]
@julienguy QL uses XSIGMA and WSIGMA rather than the XSIG and YSIG HDUs searched for in the read_xytraceset function of io/xytraceset.py. However, these HDUs (XSIGMA and WIGMA) do not contain wavemin and wavemax information, so a failure still occurs in _traceset_from_image even if the correct HDUs are found. If this information is found, QL finishes processing for arcs (and science).

**Comment by Srheft** [comment]
Running quicklook in master:

**Exact command**: 
time desi_quicklook -i qlconfig_arc.yaml -n 20191001 -c b3 -e 3573 --plots

**Exact error**: 

2018-08-20 21:43:46,660 QuickLook INFO : Running ResolutionFit
desispec.psf is DEPRECATED, PLEASE USE desispec.xytraceset
INFO:xytraceset.py:107:read_xytraceset: npix_y=4096
2018-08-20 21:43:47,279 desiutil.log.dlm58.info INFO : npix_y=4096
INFO:xytraceset.py:115:read_xytraceset: psf is a 'psf'
2018-08-20 21:43:47,280 desiutil.log.dlm58.info INFO : psf is a 'psf'
INFO:xytraceset.py:35:_traceset_from_image: read xcoef from hdu XTRACE
2018-08-20 21:43:47,280 desiutil.log.dlm58.info INFO : read xcoef from hdu XTRACE
INFO:xytraceset.py:35:_traceset_from_image: read ycoef from hdu YTRACE
2018-08-20 21:43:47,281 desiutil.log.dlm58.info INFO : read ycoef from hdu YTRACE
INFO:xytraceset.py:140:read_xytraceset: wavemin=3533.0 wavemax=5998.0
2018-08-20 21:43:47,281 desiutil.log.dlm58.info INFO : wavemin=3533.0 wavemax=5998.0
2018-08-20 21:43:47,541 QuickLook CRITICAL : Failed to run PA ResolutionFit error was no ysig coefficents were read in the PSF
Traceback (most recent call last):
  File "/desi/software/desispec/py/desispec/quicklook/quicklook.py", line 285, in runpipeline
    inp=pa(inp,**pargs)
  File "/desi/software/desispec/py/desispec/quicklook/pas.py", line 18, in __call__
    return self.run(*args,**kwargs)
  File "/desi/software/desispec/py/desispec/quicklook/procalgs.py", line 825, in run
    return self.run_pa(input_frame,psfinfile,psfoutfile,usesigma,linelist=linelist,npoly=npoly,nbins=nbins,domain=domain)
  File "/desi/software/desispec/py/desispec/quicklook/procalgs.py", line 841, in run_pa
    input_frame.resolution_data=get_resolution(input_frame.wave,input_frame.nspec,newpsf,usesigma=usesigma)
  File "/desi/software/desispec/py/desispec/quicklook/palib.py", line 134, in get_resolution
    thissigma=psf.ysigma(ispec,wave) #- in pixel units
  File "/desi/software/desispec/py/desispec/quicklook/qlpsf.py", line 130, in ysigma
    return self.traceset.ysig_vs_wave(ispec,wave)
  File "/desi/software/desispec/py/desispec/xytraceset.py", line 61, in ysig_vs_wave
    raise RuntimeError("no ysig coefficents were read in the PSF")
RuntimeError: no ysig coefficents were read in the PSF
Failed to run PA ResolutionFit

--------------------------------------------------------
Data on NERSC:
**ARC exposure**: /global/project/projectdirs/desi/spectro/redux/QL/sarahE/exposures/20191001/00003573
**Calibration data**: /global/project/projectdirs/desi/spectro/ccd_calibration_data/trunk/SIM/psf-b3.fits

---------------------------------------------------------
**Env. vars**: 
export DESI_ROOT=$HOME/desi
export DESI_SPECTRO_SIM=$DESI_ROOT/spectro/sim
export DESI_SPECTRO_REDUX=$DESI_ROOT/spectro/redux
export DESI_SPECTRO_DATA=$DESI_ROOT/spectro/data
export QL_SPEC_REDUX=$DESI_SPECTRO_REDUX/QL/
export QL_SPEC_DATA=$DESI_SPECTRO_DATA/QL/
export SIM_DIR=$DESI_SPECTRO_SIM/$PIXPROD
export REDUX_DIR=$DESI_SPECTRO_REDUX/$SPECPROD
export DESI_CCD_CALIBRATION_DATA=$HOME/desi/ccd_calibration_data
export QL_CALIB_DIR=$HOME/desi/ql_calib
export QL_CONFIG_DIR=$HOME/desi/run_quicklook/config_files

**Comment by sbailey** [comment]
Hi Sarah.  These instructions point to `ql_calib` and `config_files` that are in your NERSC $HOME directory which isn't readable by others so we can't run with the exact same configuration as you for debugging.  For those of us who don't run quicklook daily, could you please move these to /project/projectdirs/desi/srheft and run `chgrp -R desi /project/projectdirs/desi/srheft` and `chmod -R g+rX /project/projectdirs/desi/srheft` on that?  Thanks.

**Comment by Srheft** [comment]
Done. 

ARC exposure: /project/projectdirs/desi/**users**/srheft/spectro/redux/QL/exposures/20191001/00003573
Calibration data: /project/projectdirs/desi/spectro/ccd_calibration_data/trunk/SIM/psf-b3.fits

**Comment by julienguy** [comment]
Can you please tell me where to find the qlconfig_arc.yaml file at NERSC?
I can't find it in  /project/projectdirs/desi/users/srheft/

**Comment by sbailey** [comment]
Does quicklook use $QL_CALIB_DIR anymore?  That also appears to be missing from `/project/projectdirs/desi/users/srheft/`.  I thought that was a temporary workaround for psf and fiberflat files that was supposed to be superseded by getting quicklook and $DESI_CCD_CALIBRATION_DATA formats in sync with each other.

By requesting "cut-and-paste instructions" I literally mean something that any of us could cut-and-paste to go from a fresh NERSC login and end up with the reported error, e.g.
```
#- configure the master copy of all code
source /project/projectdirs/desi/software/desi_environment.sh master

# TODO: insert environment variable definitions here pointing to directories
# that everyone can see

# TODO: cd somewhere so that it can find `config_files/qlconfig_arc.yaml`
# in the following command.
# e.g. a directory copy of ~srheft/desi/run_quicklook/ perhaps

#- run quicklook
time desi_quicklook -i ./config_files/qlconfig_arc.yaml -n 20191001 -c b3 -e 3573 --plots
```
Thanks for the help.

**Comment by julienguy** [comment]
I can reproduce the feature with the default config file.
Working on the I/O ...

**Comment by julienguy** [comment]
I fixed the issue, but I am now replacing the qlpsf by xytraceset everywhere in QL to avoid future inconsistencies. It is not easy to navigate in the code (in particular when a variable named 'file_psf' is actually a class instance) ...

**Comment by julienguy** [comment]
and each test takes 4minutes to run.

**Comment by sbailey** [comment]
Fixed by PR #682, which uses `xytraceset` throughout both QL and offline instead of having similar but different classes with inconsistencies between the two.
